test/AutoDiff/Serialization/derivative_attr.swift

```diff
@@ -0,0 +1,27 @@
+// RUN: %empty-directory(%t)
```

Comment:
Note: https://github.com/apple/swift/pull/28781 added a a file of the same name (`test/AutoDiff/Serialization/derivative_attr.swift`) to `master` branch. It covers the functionality added in this test: `@derivative` attribute serialization for `init` and `subscript`.

---

So far, `AutoDiff` tests on `master` branch and `tensorflow` branch have been kept separate:
* `master` branch tests are all categorized in `test/AutoDiff/XXX` subdirectories, like ` test/AutoDiff/Parse`.
* `tensorflow` branch tests exist in `test/AutoDiff` at the top-level. Some multi-file tests include files in `test/AutoDiff/Inputs` and/or their own subdirectories, like `test/AutoDiff/silgen_thunking/main.swift` and `swift/test/AutoDiff/Inputs/silgen_thunking_other_module.swift`.

I think separation is nice because it makes it clear which tests have been upstreamed and prevents merge conflicts.

How about moving all non-upstreamed `tensorflow` branch tests to a `test/AutoDiff/tensorflow` subdirectory? cc @rxwei @bgogul 

As AutoDiff code is upstreamed, tests can be polished and moved to `test/AutoDiff/xxx`, so eventually `test/AutoDiff/tensorflow` becomes empty. On `tensorflow` branch, tests should never be added to `test/AutoDiff` outside of `test/AutoDiff/tensorflow`. On `master` branch, `test/AutoDiff/tensorflow` does not exist and tests should be added to `test/AutoDiff/XXX`.

I'm happy to start this after this PR is merged!

Reply:
A separate folder sounds good, but perhaps instead put upstreamed ones in `test/AutoDiff/upstreamed/`?